controllers/route-types.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+function createSection({ path, controllerPath, langTitlePath }) {
+    return {
+        path,
+        controller: function(pages, sectionPath, sectionId) {
+            return require(controllerPath)(pages, sectionPath, sectionId);
+        },
+        langTitlePath,
+        pages: null
+    };
+}
+
+const defaults = {
+    isPostable: false,
+    allowQueryStrings: false,
+    live: false
+};
+
+function withDefaults(props) {
+    return Object.assign({}, defaults, props);
+}
+
+function basicRoute(props) {
+    return Object.assign(
+        {},
+        defaults,
+        {
+            live: true
+        },
+        props
+    );
+}
+
+function staticRoute(props) {
+    const staticDefaults = {
+        static: true,
+        live: true
+    };
+    return Object.assign({}, defaults, staticDefaults, props);
+}
+
+function dynamicRoute(props) {
+    const dynamicDefaults = {
+        static: false,
+        live: true
+    };
+    return Object.assign({}, defaults, dynamicDefaults, props);
+}
+
+function cmsRoute(props) {
+    const cmsDefaults = {
+        useCmsContent: true,
+        live: true
+    };
+    return Object.assign({}, defaults, cmsDefaults, props);
+}
+
+function legacyRoute(props) {
```

Comment:
Might be nice (for our own sanity) to comment each type briefly to define the distinctions?